map/bookmark_manager.cpp

```diff
@@ -2258,6 +2258,12 @@ bool BookmarkManager::HasBmCategory(kml::MarkGroupId groupId) const
   return (IsBookmarkCategory(groupId) && GetBmCategorySafe(groupId) != nullptr);
 }
 
+bool BookmarkManager::HasBookmark(kml::MarkId markId) const
+{
+  CHECK_THREAD_CHECKER(m_threadChecker, ());
+  return (IsBookmark(markId) && GetBookmark(markId) != nullptr);
+}
```

Comment:
I have doubts that this function is ok.
What if we try to reproduce this crash from a Bookmark in the middle of list? Bookmark's ids will be shifted and some other Bookmark will be displayed in PP?

I suppose that the best option here is to force close PP when editing Bookmarks.

Reply:
During the updating, the bookmarks and tracks don't change their IDs - they detached from one group and attached to the other, so this operation is safe even if I change the bookmark in the middle of the list. All the bookmarks from all groups have unique IDs.

```cpp
void BookmarkManager::MoveBookmark(kml::MarkId bmID, kml::MarkGroupId curGroupID, kml::MarkGroupId newGroupID)
{
  CHECK_THREAD_CHECKER(m_threadChecker, ());
  DetachBookmark(bmID, curGroupID);
  AttachBookmark(bmID, newGroupID);

  SetLastEditedBmCategory(newGroupID);
}
```

Check the record:
![Simulator Screen Recording - iPad Air (5th generation) - 2024-05-23 at 13 08 23](https://github.com/organicmaps/organicmaps/assets/79797627/e7ac0caf-865c-4046-876c-0e700bc4a421)

Reply:
Always incrementing during the reloading of the collection (it means delete all and recreate).

During the editing (name, color, moving between groups), the bookmarks doesn't change their ids.

Reply:
I've add the check
```cpp
bool BookmarkManager::HasBookmark(kml::MarkId markId) const
{
  CHECK_THREAD_CHECKER(m_threadChecker, ());
  CHECK(IsBookmark(markId), ());
  return (GetBookmark(markId) != nullptr);
}
```